lua/entities/gmod_wire_expression2/core/e2lib.lua

```diff
@@ -439,89 +439,114 @@ end
 -- -------------------------- disabling extensions -----------------------------
 
 do
-	local extensions = {}
-
-	if not sql.TableExists("wire_expression2_extensions") then
-		sql.Query("CREATE TABLE IF NOT EXISTS wire_expression2_extensions (name varchar(255), enabled tinyint)")
-		sql.Query("CREATE UNIQUE INDEX name ON wire_expression2_extensions(name)")
+	E2Lib.extensions = { status = {}, list = {} }
+	local sortExtensionsList = true
+
+	local list = sql.Query( "SELECT * FROM wire_expression2_extensions" )
+	if list then
+		for i = 1, #list do
+			local row = list[ i ]
+			E2Lib.extensions.status[ row.name ] = row.enabled == "1" and true or false
+			E2Lib.extensions.list[ i ] = row.name
+		end
+	else
+		sql.Query( "CREATE TABLE wire_expression2_extensions ( name VARCHAR(32), enabled BOOLEAN )" )
+		sql.Query( "CREATE UNIQUE INDEX name ON wire_expression2_extensions ( name )" )
 	end
-
-	function extensions.GetStatus(name, default)
-		local value = sql.QueryValue(string.format("SELECT enabled FROM wire_expression2_extensions WHERE (name = %s)", sql.SQLStr(name)))
-		value = value and value ~= "0" or default
-		return value
+	
+		
+	function E2Lib.RegisterExtension( name, default )
+		if E2Lib.extensions.status[ name ] == nil then
+			E2Lib.SetExtensionStatus( name, default or false )
+			E2Lib.extensions.list[ #E2Lib.extensions.list + 1 ] = name
+			sortExtensionsList = true
+		end
+		
+		-- DO NOT MODIFY THIS LINE, PLEASE!
+		assert( E2Lib.extensions.status[ name ], "Skipping disabled E2 extension '" .. name .. "'. To enable, run 'wire_expression2_extension_enable " .. name .. "'\n" )
```

Comment:
I don't agree with the notion of having unmodifiable lines in a project.
Much less if there's no reason given.